webapp/python/app.py

```diff
@@ -0,0 +1,504 @@
+
+
+@app.route("/")
+def get_index():
+    me = get_session_user()
+
+    cursor = db().cursor()
+    cursor.execute(
+        "SELECT `id`, `user_id`, `body`, `created_at`, `mime` FROM `posts` ORDER BY `created_at` DESC"
+    )
+    posts = make_posts(cursor.fetchall())
+
+    return flask.render_template("index.html", posts=posts, me=me)
+
+
+@app.route("/@<account_name>")
+def get_user_list(account_name):
+    cursor = db().cursor()
+
+    cursor.execute(
+        "SELECT * FROM `users` WHERE `account_name` = %s AND `del_flg` = 0",
+        (account_name,),
+    )
+    user = cursor.fetchone()
+    if not user:
+        flask.abort(404)  # raises exception
+
+    cursor.execute(
+        "SELECT `id`, `user_id`, `body`, `mime`, `created_at` FROM `posts` WHERE `user_id` = %s ORDER BY `created_at` DESC",
+        (user["id"],),
+    )
+    posts = make_posts(cursor.fetchall())
+
+    cursor.execute(
+        "SELECT COUNT(*) AS count FROM `comments` WHERE `user_id` = %s", (user["id"],)
+    )
+    comment_count = cursor.fetchone()["count"]
+
+    cursor.execute("SELECT `id` FROM `posts` WHERE `user_id` = %s", (user["id"],))
+    post_ids = [p["id"] for p in cursor]
+    post_count = len(post_ids)
+
+    commented_count = 0
+    if post_count > 0:
+        cursor.execute(
+            "SELECT COUNT(*) AS count FROM `comments` WHERE `post_id` IN %s",
+            (post_ids,),
+        )
+        commented_count = cursor.fetchone()["count"]
+
+    me = get_session_user()
+
+    return flask.render_template(
+        "user.html",
+        posts=posts,
+        user=user,
+        post_count=post_count,
+        comment_count=comment_count,
+        commented_count=commented_count,
+        me=me,
+    )
+
+
+def _parse_iso8601(s):
+    # http://bugs.python.org/issue15873
+    # Ignore timezone
+    m = re.match(r"(\d{4})-(\d{2})-(\d{2})[ tT](\d{2}):(\d{2}):(\d{2}).*", s)
+    if not m:
+        raise ValueError("Invlaid iso8601 format: %r" % (s,))
+    return datetime.datetime(*map(int, m.groups()))
+
+
+@app.route("/posts")
+def get_posts():
+    cursor = db().cursor()
+    max_created_at = flask.request.args["max_created_at"] or None
+    if max_created_at:
+        max_created_at = _parse_iso8601(max_created_at)
+        cursor.execute(
+            "SELECT `id`, `user_id`, `body`, `mime`, `created_at` FROM `posts` WHERE `created_at` <= %s ORDER BY `created_at` DESC",
+            (max_created_at,),
+        )
+    else:
+        cursor.execute(
+            "SELECT `id`, `user_id`, `body`, `mime`, `created_at` FROM `posts` WHERE ORDER BY `created_at` DESC"
+        )
+    results = cursor.fetchall()
+    posts = make_posts(results)
+    return flask.render_template("posts.html", posts=posts)
+
+
+@app.route("/posts/<id>")
+def get_posts_id(id):
+    cursor = db().cursor()
+
+    cursor.execute("SELECT * FROM `posts` WHERE `id` = %s", (id,))
+    posts = make_posts(cursor.fetchall(), all_comments=True)
+    if not posts:
```

Comment:
ruby や go と合わせて、`len` で長さを取る方が良いかと思いました

```suggestion
    if len(posts) == 0:
```

Reply:
それは言語ごとの構文やイディオムの違いなのでこのままでいいと思います。PEP8でもこうなっています。

> <img width="672" alt="image" src="https://github.com/user-attachments/assets/d10843ef-9a21-42c1-9442-5c6461e42162" />
>
> -- https://peps.python.org/pep-0008/#programming-recommendations

Reply:
各言語で一番自然な書き方が良いです

Reply:
> 各言語で一番自然な書き方が良いです

承知しました。それであればこのままで大丈夫です！